lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -714,35 +716,51 @@ def wait_for_host_in_dashboard(host)
 
         # Ensure the host has requested a cert, then sign it
         #
-        # @param [Host, Array<Host>, String, Symbol] host    One or more hosts to act upon,
+        # @param [Host, Array<Host>, String, Symbol] host   One or more hosts to validate signing certificate,
         #                            or a role (String or Symbol) that identifies one or more hosts.
+        #                            No argument, or an empty array means no validation of success
+        #                            for specific hosts will be performed.
         #
         # @return nil
         # @raise [FailTest] if process times out
-        def sign_certificate_for(host)
-          block_on host do | host |
-            if [master, dashboard, database].include? host
+        def sign_certificate_for(host = [])
+          hostnames = []
+          hosts = host.is_a?(Array) ? host : [host]
+          hosts.each{ |current_host|
+            if [master, dashboard, database].include? current_host
 
-              on host, puppet( 'agent -t' ), :acceptable_exit_codes => [0,1,2]
-              on master, puppet( "cert --allow-dns-alt-names sign #{host}" ), :acceptable_exit_codes => [0,24]
+              on current_host, puppet( 'agent -t' ), :acceptable_exit_codes => [0,1,2]
+              on master, puppet( "cert --allow-dns-alt-names sign #{current_host}" ), :acceptable_exit_codes => [0,24]
```

Comment:
Can this line end up being done in parallel? The master might not like trying to do several signings at once if it's locking files in the ssl dir or elsewhere.